lightning/src/chain/keysinterface.rs

```diff
@@ -821,29 +814,14 @@ impl KeysInterface for KeysManager {
 		self.derive_channel_keys(channel_value_satoshis, ix_and_nanos, self.starting_time_secs)
 	}
 
-	fn get_onion_rand(&self) -> (SecretKey, [u8; 32]) {
+	fn get_secure_random_bytes(&self) -> [u8; 32] {
 		let mut sha = self.derive_unique_start();
 
-		let child_ix = self.session_child_index.fetch_add(1, Ordering::AcqRel);
-		let child_privkey = self.session_master_key.ckd_priv(&self.secp_ctx, ChildNumber::from_hardened_idx(child_ix as u32).expect("key space exhausted")).expect("Your RNG is busted");
-		sha.input(&child_privkey.private_key.key[..]);
-
-		let mut rng_seed = sha.clone();
-		// Not exactly the most ideal construction, but the second value will get fed into
-		// ChaCha so it is another step harder to break.
-		rng_seed.input(b"RNG Seed Salt");
-		sha.input(b"Session Key Salt");
-		(SecretKey::from_slice(&Sha256::from_engine(sha).into_inner()).expect("Your RNG is busted"),
-		Sha256::from_engine(rng_seed).into_inner())
-	}
-
-	fn get_channel_id(&self) -> [u8; 32] {
-		let mut sha = self.derive_unique_start();
-
-		let child_ix = self.channel_id_child_index.fetch_add(1, Ordering::AcqRel);
-		let child_privkey = self.channel_id_master_key.ckd_priv(&self.secp_ctx, ChildNumber::from_hardened_idx(child_ix as u32).expect("key space exhausted")).expect("Your RNG is busted");
+		let child_ix = self.rand_bytes_child_index.fetch_add(1, Ordering::AcqRel);
+		let child_privkey = self.rand_bytes_master_key.ckd_priv(&self.secp_ctx, ChildNumber::from_hardened_idx(child_ix as u32).expect("key space exhausted")).expect("Your RNG is busted");
```

Comment:
"Your CSRNG is busted" ?

Reply:
We have "RNG is busted" in a ton of places. I don't think its really all that critical to explicitly say CSRNG in a panic message (which even a biased RNG should suffice to not hit), as long as the docs (which hopefully is the only thing users ever see) are good.